File: src/main/java/com/bourgeolet/task_manager/entity/Task.java
package com.bourgeolet.task_manager.entity;

import com.bourgeolet.task_manager.model.task.TaskStatus;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Entity
@Table(name = "tasks")
@Getter
@Setter
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "task_seq")
    @SequenceGenerator(name = "task_seq", sequenceName = "task_seq")
    private Long id;

    @ManyToOne(cascade = CascadeType.DETACH)
    @JoinColumn(name = "user_username")
    private User user;

    private String title;

    private String description;

    private int priority;

    private List<String> tags;

    @Enumerated(EnumType.STRING)
    private TaskStatus status = TaskStatus.TODO;
}

File: frontend/src/app/features/task/models/task.ts
import { ColumnId } from "../../../../environments/type";
import { Tag } from "../../tag/models/tag";

export interface Task {
  id: number;
  title: string;
  description?: string;
  userAffectee?: string | null;
  priority?: number;
  tags?: Tag[] | null;
  status?: ColumnId;
}


